### PR TITLE
Update BLE HAL doxygen

### PR DESCRIPTION
**Pull request by doujiahu**
Update BLE HAL doxygen

Description
-----------
Update BLE HAL doxygen
Remove unnecessary extern in front of function prototype

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/ble_hal/include/bt_hal_manager.h [hunk 1]
```diff
@@ -59,16 +59,10 @@
...
- * @brief  Help functions to convert version to string.
- */
-#define STR_HELPER( x, y, z )    # x "." # y "." # z
-#define STR( x, y, z )           STR_HELPER( x, y, z )
-
 /**
  * @brief  Stringified version number.
  */
-#define btSTRING_VERSION                        STR( btMAJOR_VERSION, btMINOR_VERSION, btPATCH_VERSION )
+#define btSTRING_VERSION                        # btMAJOR_VERSION "." # btMINOR_VERSION "." # btPATCH_VERSION
```
Comment (ravibhagavandas) [inline libraries/abstractions/ble_hal/include/bt_hal_manager.h:65]: Can you confirm this works ? The stringification of macro requires it to be passed as an argument and requires one level of indirection as mentioned here:
https://gcc.gnu.org/onlinedocs/gcc-4.8.5/cpp/Stringification.html

Comment (doujiahu) [reply libraries/abstractions/ble_hal/include/bt_hal_manager.h]: Added back macros and appended "bt" in front of macro to address ACS team concerns

**Pull request merged**